admin-notice/js/script.js

```diff
@@ -21,19 +14,18 @@ function persistDismissedNotice(containerElement) {
 	}
 }
 
-function onDismissed(event) {
-	const noticeContainer = tryGetNoticeContainer(event && event.target);
-	if (noticeContainer) {
-		persistDismissedNotice(noticeContainer);
+function vipNoticeClickHandler( ev ) {
+	const button = ev.target.closest( '.notice-dismiss' );
+	if ( button ) {
+		const noticeContainer = button.closest( `[data="${dismissal_data.data_attribute}"]` );
+		if (noticeContainer) {
+			persistDismissedNotice(noticeContainer);
+		}
 	}
 }
 
 function registerDismissHooks() {
-	const notices = document.querySelectorAll( '.vip-notice .notice-dismiss' );
-
-	for ( const notice of notices ) {
-		notice.addEventListener( "click", onDismissed );
-	}
+	document.querySelectorAll( '.vip-notice' ).forEach( notice => notice.addEventListener( 'click', vipNoticeClickHandler ) );
```

Comment:
Because events bubble, we can set up an event listener on a parent element and then check which child was clicked.

This also prevents a race condition when we set up event listeners but the dismiss button is not there yet.